pkg/controller/deployment/deployment_controller.go

```diff
@@ -173,6 +173,11 @@ func (dc *DeploymentController) addDeployment(obj interface{}) {
 func (dc *DeploymentController) updateDeployment(old, cur interface{}) {
 	oldD := old.(*apps.Deployment)
 	curD := cur.(*apps.Deployment)
+	if oldD.ResourceVersion == curD.ResourceVersion {
+		// Periodic resync will send update events for all known deployments.
+		// Two different versions of the same deployment will always have different RVs.
+		return
+	}
```

Comment:
for replicaSet this check exists, but i am not seeing this pattern on the main object, the same checks are missing from CronJob and StatefulSet, etc. I am guessing there is a good reason for it. @kow3ns @deads2k  is this necessary on the main object ?

Reply:
I think this may just be an oversight, compare ResourceVersion process is just used in the case of two re-list result not same. If my guess is right, should I add the same check for CronJob and StatefulSet?

Reply:
Basically the main issue can be showcased here: https://github.com/kubernetes/kubernetes/blob/61b3c028ba618a939559c39befb546ae5e5fd0b9/pkg/controller/deployment/deployment_controller.go#L489-L497 

Suppose something doesn't work  in our single sync iteration (eg. we are unable to create a ReplicaSet, etc.. (all errors bubble up here)) we will stop trying after `maxRetries`. That might change and we might succeed in the future, thus the resync will ensure that we will always retry even if the item drops out of the queue.

Also the `maxRetries` is not that important because with exponential rate limiter you will get into unusable long sync intervals after some time and the 10min resync will help here.

You can see this check on secondary informers (eg. ReplicaSet in deployment_controller), but these are only responsible for enqueuing the Deployment and we don't need to enqueue the Deployment multiple times due to the mechanics described above.

Conclusion: we should not have this check on the main informer/object